SBNOpenDiskTree(): make it work with node descriptors with non-increasing nBinStart

rouault opened the pull request:
Fixes https://github.com/OSGeo/shapelib/issues/106 / https://github.com/OSGeo/gdal/issues/9430

thbeu commented:
Would be useful to have at least some minimal test code for the SBN API in the repo. Currently there is none. See https://github.com/OSGeo/shapelib/issues/106#issuecomment-1986799495 for an example.

rouault commented:
> Would be useful to have at least some minimal test code for the SBN API in the repo

done

thbeu commented:
> > Would be useful to have at least some minimal test code for the SBN API in the repo
> 
> done

Please add all the new files to EXTRA_DIST in Makefile.am

rouault commented:
> Please add all the new files to EXTRA_DIST in Makefile.am

done. shapelib.def was also actually missing. I've fixed that. I've also modified the linux-autoconf CI config to actually do a cmake build on the tarball generated by makedist...

thbeu commented:
> > Please add all the new files to EXTRA_DIST in Makefile.am
> 
> done. shapelib.def was also actually missing. I've fixed that. I've also modified the linux-autoconf CI config to actually do a cmake build on the tarball generated by makedist...

Very good. Thanks a lot!
